File: 1167-minimum-cost-to-connect-sticks/1167-minimum-cost-to-connect-sticks.cpp
class Solution {
public:
    int connectSticks(vector<int>& sticks) {
        
          priority_queue<int,vector<int>,greater<int>>pq(sticks.begin(),sticks.end());
          int sum=0;
          while(pq.size()>=2){
              
                  int val=pq.top();
                  pq.pop();
                  val+=pq.top();
                  pq.pop();
                  sum+=val;
                  pq.push(val);
                  val=0;
              
          }
        
        
          return sum;
            
          
            
    }
};

File: 1167-minimum-cost-to-connect-sticks/1167-minimum-cost-to-connect-sticks.py
class Solution:
    
    def connectSticks(self, sticks: List[int]) -> int:
        
        
        heapq.heapify(sticks)
        sum1=0
        while len(sticks)>1:
                
                a=heapq.heappop(sticks)
                b=heapq.heappop(sticks)
                print(a+b)
                sum1+=a+b
                
                heapq.heappush(sticks,(a+b))
                
        
        return sum1      
                
                
                
        